nexus_constructor/json_connector.py

```diff
@@ -32,23 +32,32 @@ def __init__(self):
         with open('Instrument.schema.json') as file:
             self.schema = json.load(file)
 
-    @Slot(QUrl, 'QVariant')
+    @Slot(QUrl, 'QVariant', result=bool)
     def load_file_into_instrument_model(self, file_url: QUrl, model: InstrumentModel):
         filename = file_url.toString(options=QUrl.PreferLocalFile)
         with open(filename, 'r') as file:
             json_string = file.read()
-        data = json.loads(json_string)
+
+        try:
```

Comment:
If this line onwards was extracted to a separate method then it could be unit tested by passing json strings.